src/adapters/adapter.ts

```diff
@@ -1,12 +1,19 @@
 import { DebtOrder, DebtRegistryEntry } from "../types";
+import { CollateralizedSimpleInterestLoanOrder } from "./collateralized_simple_interest_loan_adapter";
+import { SimpleInterestLoanOrder } from "./simple_interest_loan_adapter";
 
 export namespace Adapter {
     export interface Interface {
-        fromDebtOrder: (debtOrder: DebtOrder.Instance) => Promise<object>;
+        fromDebtOrder: (
+            debtOrder: DebtOrder.Instance,
+        ) => Promise<SimpleInterestLoanOrder | CollateralizedSimpleInterestLoanOrder>;
         toDebtOrder: (params: object) => Promise<DebtOrder.Instance>;
         fromDebtRegistryEntry: (entry: DebtRegistryEntry) => Promise<object>;
-        getRepaymentSchedule: (entry: DebtRegistryEntry) => Array<number>;
+        getRepaymentSchedule: (entry: DebtRegistryEntry) => number[];
         unpackParameters: (packedParams: string) => object;
+        validateAsync: (
```

Comment:
what are you our thoughts about validation APIs returning a boolean instead of void.

Reply:
Agreed